kratos/utilities/math_utils.h

```diff
@@ -56,7 +56,6 @@ namespace Kratos
  * @author Riccardo Rossi
  * @author Pooyan Dadvand
  */
-template<class TDataType>
 class MathUtils
```

Comment:
```suggestion
template<class TDataType=double>
class MathUtils
```

this way it is backward compatible, and you can replace `MathUtils<double>` => `MathUtils` one-by-one

see https://github.com/KratosMultiphysics/Kratos/blob/4920be1d8b4d3d97f4a992ef57673d96eb2bf844/kratos/utilities/parallel_utilities.h#L406

Reply:
I agree. If we can split this up in smaller (more manageable) pieces, that would be very helpful for the review process. Also, I prefer to have logically distinct changes (e.g. replacing `MathUtils<>::Min` by `std::min`) in separate PRs. Having said that, those are my preferences, but perhaps it's not practical here since the work has already been done...?

Reply:
I will separate the other methods

Reply:
I think that if you put a default template argument you need to change for MathUtils<> anyway, so it will be the same effect

Reply:
> you need to change for MathUtils<> anyway

since C++17 you don't ;)

Reply:
![](https://media.tenor.com/DJhvQ2QAttwAAAAM/monkey-shook.gif)

Reply:
Done

Reply:
This is a game changer for this sort of things. Thanks for the hint @philbucher.